src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java

```diff
@@ -887,6 +920,72 @@ public void testGetImportedKeysDW() throws SQLException {
+            getColumnMetaDataClass.put(DATA_TYPE, Integer.class);
+            getColumnMetaDataClass.put(TYPE_NAME, String.class);
+            getColumnMetaDataClass.put(COLUMN_SIZE, Integer.class);
+            getColumnMetaDataClass.put(BUFFER_LENGTH, Integer.class); // Not used
+            getColumnMetaDataClass.put(DECIMAL_DIGITS, Integer.class);
+            getColumnMetaDataClass.put(NUM_PREC_RADIX, Integer.class);
+            getColumnMetaDataClass.put(NULLABLE, Integer.class);
+            getColumnMetaDataClass.put(REMARKS, String.class);
+            getColumnMetaDataClass.put(COLUMN_DEF, String.class);
+            getColumnMetaDataClass.put(SQL_DATA_TYPE, Integer.class);
+            getColumnMetaDataClass.put(SQL_DATETIME_SUB, Integer.class);
+            getColumnMetaDataClass.put(CHAR_OCTET_LENGTH, Integer.class);
+            getColumnMetaDataClass.put(ORDINAL_POSITION, Integer.class);
+            getColumnMetaDataClass.put(IS_NULLABLE, String.class);
+            getColumnMetaDataClass.put(SCOPE_CATALOG, String.class);
+            getColumnMetaDataClass.put(SCOPE_SCHEMA, String.class);
+            getColumnMetaDataClass.put(SCOPE_TABLE, String.class);
+            getColumnMetaDataClass.put(SOURCE_DATA_TYPE, Short.class);
+            getColumnMetaDataClass.put(IS_AUTOINCREMENT, String.class);
+            getColumnMetaDataClass.put(IS_GENERATEDCOLUMN, String.class);
+            getColumnMetaDataClass.put(SS_IS_SPARSE, Short.class);
+            getColumnMetaDataClass.put(SS_IS_COLUMN_SET, Short.class);
+            getColumnMetaDataClass.put(SS_UDT_CATALOG_NAME, String.class);
+            getColumnMetaDataClass.put(SS_UDT_SCHEMA_NAME, String.class);
+            getColumnMetaDataClass.put(SS_UDT_ASSEMBLY_TYPE_NAME, String.class);
+            getColumnMetaDataClass.put(SS_XML_SCHEMACOLLECTION_CATALOG_NAME, String.class);
+            getColumnMetaDataClass.put(SS_XML_SCHEMACOLLECTION_SCHEMA_NAME, String.class);
+            getColumnMetaDataClass.put(SS_XML_SCHEMACOLLECTION_NAME, String.class);
+        }
+
+        try (Connection conn = getConnection()) {
+            ResultSetMetaData metadata = conn.getMetaData().getColumns(null, null, tableName, null).getMetaData();
+
+            // Ensure that there is an expected class for every column in the metadata result set
+            assertEquals(metadata.getColumnCount(), getColumnMetaDataClass.size());
+
+            for (int i = 1; i < metadata.getColumnCount(); i++) {
+                String columnLabel = metadata.getColumnLabel(i);
+                String columnClassName = metadata.getColumnClassName(i);
+                Class<?> expectedClass = getColumnMetaDataClass.get(columnLabel);
+
+                // Ensure the metadata column is in the metadata column class map
+                assertNotNull(expectedClass);
+
+                // Ensure the actual and expected column metadata types match
+                if (columnClassName != expectedClass.getName()) {
```

Comment:
Please use .equals() instead of == or !=
```suggestion
                if (!columnClassName.equals(expectedClass.getName())) {
```

Reply:
Completed

Reply:
Done